src/components/Avatar/tests/Avatar.test.tsx

```diff
@@ -0,0 +1,124 @@
+import {mountWithAppProvider, trigger} from 'test-utilities';
+import {Image} from 'components';
+import Avatar from '../Avatar';
+
+const actualIsServer = targets.isServer;
+
+function mockIsServer(value: boolean) {
+  (targets as any).isServer = value;
+}
+
+describe('<Avatar />', () => {
+  describe('intials', () => {
+    it('renders intials if the image is not provided', () => {
+      const avatar = mountWithAppProvider(<Avatar initials="DL" />);
+      expect(avatar.find('span[role="img"] span svg')).toHaveLength(1);
+    });
+  });
+
+  describe('source', () => {
+    afterEach(() => {
+      mockIsServer(actualIsServer);
+    });
+    it('renders an Image component with the Avatar source if one is provided', () => {
+      const src = 'image/path/';
+      const avatar = mountWithAppProvider(<Avatar source={src} />);
+      const image = avatar.find(Image);
+      expect(image.prop('source')).toBe(src);
+    });
+
+    it('safely updates', () => {
+      const src = 'image/path/';
+      const avatar = mountWithAppProvider(<Avatar source={src} />);
+      expect(() => {
+        avatar.setProps({source: 'image/new/path'});
+      }).not.toThrow();
+    });
+
+    it('does not render an Image on the server', () => {
+      mockIsServer(true);
+      const src = 'image/path/';
+      const avatar = mountWithAppProvider(<Avatar source={src} />);
+      expect(avatar.find(Image)).toHaveLength(0);
+    });
+  });
+
+  describe('customer', () => {
+    it('renders an Image component with a customer Avatar if the customer prop is true', () => {
+      const avatar = mountWithAppProvider(<Avatar customer />);
+      const image = avatar.find(Image);
+      expect(image.prop('source')).toContain('avatar-');
+    });
+
+    it('does not render a customer Avatar if a source is provided', () => {
+      const src = 'image/path/';
+      const avatar = mountWithAppProvider(<Avatar customer source={src} />);
+      const image = avatar.find(Image);
+      expect(image.prop('source')).not.toContain('avatar-');
+    });
+  });
+
+  describe('on Error with Initials', () => {
+    it('renders initials if the Image onError prop is triggered and the Intials are provided', () => {
+      const src = 'image/path/';
+      const avatar = mountWithAppProvider(
+        <Avatar size="large" initials="DL" source={src} />,
+      );
+      expect(avatar.find('span[role="img"] span svg')).toHaveLength(0);
+      trigger(avatar.find(Image), 'onError');
+      expect(avatar.find('span[role="img"] span svg')).toHaveLength(1);
+    });
+  });
+
+  describe('on Error with changed props', () => {
+    it('re-renders the image if a the source prop is changed after an error', () => {
+      const src = 'image/path/';
+      const workingSrc = 'image/goodPath/';
+      const avatar = mountWithAppProvider(
+        <Avatar size="large" initials="DL" source={src} />,
+      );
+      trigger(avatar.find(Image), 'onError');
```

Comment:
```suggestion
      trigger(avatar.find(Image), 'onError');
      expect(avatar.find(Image)).toHaveLength(0);
```

I feel like we should add this check that the image indeed does not render after an error then re-renders

Reply:
Also is there a preference for `trigger` vs `simulate`?